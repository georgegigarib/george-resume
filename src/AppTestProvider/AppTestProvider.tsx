import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
