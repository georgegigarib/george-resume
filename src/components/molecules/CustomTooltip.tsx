import React from 'react'
import { Tooltip, TooltipProps } from '@mui/material'
import clsx from 'clsx'

const CustomTooltip = ({
  title,
  bgColor = 'bg-gray-700',
  textSize = 'text-md',
  className = '',
  children,
  ...props
}: TooltipProps & { bgColor?: string; textSize?: string; className?: string }) => {
  const stopPropagation = (e: React.MouseEvent) => {
    e.stopPropagation()
  }

  return (
    <Tooltip
      title={
        <div onClick={stopPropagation} className={clsx('overflow-auto max-h-[100px] rounded-md', bgColor, textSize)}>
          {title}
        </div>
      }
      {...props}
    >
      <div className={className}>{children}</div>
    </Tooltip>
  )
}

export default CustomTooltip
