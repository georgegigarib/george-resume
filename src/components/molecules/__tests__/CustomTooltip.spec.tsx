import { render, screen, fireEvent } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import CustomTooltip from '@/components/molecules/CustomTooltip'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

describe('CustomTooltip component', () => {
  const renderComponent = (title: string, children: React.ReactElement, props = {}) =>
    render(
      <AppTestProvider>
        <CustomTooltip title={title} {...props}>
          {children}
        </CustomTooltip>
      </AppTestProvider>
    )

  it('matches the snapshot', () => {
    const { asFragment } = renderComponent('Tooltip title', <span>Hover me</span>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the tooltip title on hover', async () => {
    renderComponent('Tooltip title', <span>Hover me</span>)
    fireEvent.mouseOver(screen.getByText('Hover me'))

    expect(await screen.findByText('Tooltip title')).toBeInTheDocument()
  })

  it('removes tooltip message when mouse leave', async () => {
    renderComponent('Tooltip title', <span>Hover me</span>)
    fireEvent.mouseOver(screen.getByText('Hover me'))

    expect(await screen.findByText('Tooltip title')).toBeInTheDocument()

    fireEvent.mouseLeave(screen.getByText('Hover me'))

    //TODO: Fix this test, it works, but it's not the best way to test it
    setTimeout(() => {
      expect(screen.queryByText('Tooltip title')).not.toBeInTheDocument()
    }, 0)
  })
})
