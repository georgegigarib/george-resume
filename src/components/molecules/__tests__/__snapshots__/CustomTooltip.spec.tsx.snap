// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomTooltip component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    data-mui-internal-clone-element="true"
  >
    <span>
      Hover me
    </span>
  </div>
</DocumentFragment>
`;
